### DISCUSSION TITLE
more `node:crypto`

### DISCUSSION THREAD

**Discussion by demosjarco** [cloudflare/workerd#2692] [Node.js APIs]
From `node:crypto`:
- `createECDH`
- `createCipher`
- `createCipheriv`
- `createDecipher`
- `createDecipheriv`
- `createHash`
- `createHmac`

I've confirmed these are missing upstream from `uncrypto` (inside `unenv`)

I'm looking to do ECIES with the [`standard-ecies`](https://www.npmjs.com/package/standard-ecies) package and it uses these crypto calls under the hood.

**Comment by jasnell** [comment]
These are coming. No specific timeline yet as it's a very large project and we're working to also do some cleanup in node.js to make it the task easier. It'll take some time.

**Comment by ragelo** [comment]
Got `[unenv] crypto.createSign is not implemented yet!` issue when tried to use `JWT`  from `google-auth-library`.
It's missing in the first message, so posting here for visibility.

**Comment by cazcik** [comment]
obviously `@opennextjs/cloudflare` is still in beta, but getting `Error: [unenv] crypto.createCipheriv is not implemented yet!` when using `next-auth`.

```
"next": "14.2.16"
"next-auth": "5.0.0-beta.25"
"@opennextjs/cloudflare": "^0.2.1"
"wrangler": "^3.86.0"

compatibility_date = "2024-09-26"
compatibility_flags = ["nodejs_compat"]
```

**Reply by pantharshit007** [reply]
did you found any workaround or solution to this?

**Reply by saamdotexe** [reply]
I just encountered this issue, hopefully we'll have more news on it soon.

**Comment by skyfallwastaken** [comment]
The lack of these functions means that Auth.js seems to be completely broken on Pages (at least with SvelteKit) - would be great if this was resolved :)

**Reply by saamdotexe** [reply]
That was my issue too, consider switching to BetterAuth, it's a great alternative

**Reply by skyfallwastaken** [reply]
Will check it out, thanks!

**Comment by alinalihassan** [comment]
I'm also having issues with `[unenv] crypto.createSign is not implemented yet!`. My use case can't easily avoid the lack of crypto functions
